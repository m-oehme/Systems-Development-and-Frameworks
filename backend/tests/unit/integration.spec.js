const { gql } = require("apollo-server");
const { createTestClient } = require("apollo-server-testing");

const { constructTestServer } = require("./__utils");

const GET_HELLO = gql`
  query {
    message: hello
  }
`;

// const GET_TODOS = gql`
//     query {
//         todo: todo
//     }
// `;

describe("Testing Querys", () => {
  it("receiving hello world message", async () => {
    const { testServer } = constructTestServer();

    // use the test server to create a query function
    const { query } = createTestClient(testServer);

    const res = await query({ query: GET_HELLO, variables: { id: 1 } });
    expect(res).toMatchSnapshot();
  });

  it.todo("receiving todolist response");
});
