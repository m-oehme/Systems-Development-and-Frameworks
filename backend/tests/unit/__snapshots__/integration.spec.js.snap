// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Querys receiving hello world message 1`] = `
Object {
  "data": Object {
    "message": "I am coming from the Backend to kill you! NOT",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
